src/commands/wait_for_job.yml

```diff
@@ -46,17 +46,14 @@ steps:
           exit 0;
         fi
 
-        api_endpoint="api/v2/workflow/${CIRCLE_WORKFLOW_ID}/job"
-
         mkdir -p /tmp/swissknife
 
         # This is a global variable used to get return value for get_job_status
         job_status=""
         job_number=""
 
         get_job_status() {
-          wf_url="https://circleci.com/$api_endpoint?circle-token=${CIRCLE_TOKEN}"
-          curl -f -s $wf_url > /tmp/swissknife/wf_$CIRCLE_WORKFLOW_ID.json
+          curl --header 'authorization: Basic ${CIRCLE_TOKEN}' -f -s https://circleci.com/api/v2/workflow/${CIRCLE_WORKFLOW_ID}/job > /tmp/swissknife/wf_$CIRCLE_WORKFLOW_ID.json
```

Comment:
if we are changing this you need to use
```
Circle-Token
```
as the header name not basic i believe.

also this orb is used in production and works right now. interesting to see that you get not found 🤔

Reply:
I used the proposed method in documention here https://circleci.com/docs/api/v2/#operation/listWorkflowJobs
<img width="457" alt="Screenshot 2022-02-11 at 22 38 58" src="https://user-images.githubusercontent.com/1590371/153673946-f0f55593-5192-49ff-8ce1-7c04739bb73a.png">

I'll try again next week and let you know, could it be linked somehow to using a different version of pipelines? 
I was using `version: 2.1`.

Reply:
ah yeah the circleci docs arent great at the auth part. this PR will break the orb :)

Reply:
Sorry my message wasn't very clear, based on documentation there are three possible authentication methods:
- `api_key_header` using `Circle-Token` header key
- `basic_auth` using basic authentication (the one I implemented in this PR)
- `api_key_query ` using `circle-token` key in query parameters (not in headers), which is the method your are currently using, and which will be deprecated at some point 
<img width="570" alt="Screenshot 2022-02-12 at 22 59 47" src="https://user-images.githubusercontent.com/1590371/153729861-4a2e54d7-b9b5-490d-9bcc-e225af1f7212.png">

I used `basic_auth` but `api_key_header` would work fine as well :)

Note: before making this PR I launched a CircleCi build with SSH session to test it, and it works for me, so I don't think it would break the orb (or is there something I missed 🤔 ?):
<img width="1154" alt="Screenshot 2022-02-12 at 22 59 16" src="https://user-images.githubusercontent.com/1590371/153729971-cb142566-d708-4424-9d82-232791d63b11.png">

For some reason, in my case the `api_key_query` method doesn't work though.
<img width="841" alt="Screenshot 2022-02-12 at 22 58 43" src="https://user-images.githubusercontent.com/1590371/153730000-a29d6c79-7b9e-4a9e-bac3-8922fd1fb77f.png">

This error wasn't covered and resulted in a 22 error code:
<img width="607" alt="Screenshot 2022-02-12 at 22 56 15" src="https://user-images.githubusercontent.com/1590371/153730027-e8565c50-9c83-473d-8c9f-416d1a8e4030.png">
I investigated via SSH to understand where the issue came from, and to me it seems related to authentication method.

By the way, thanks a lot for making this orb, I don't understand how this feature isn't built-in in circleci, your orb is really appreciated :)

Cheers!

Reply:
note: I just noticed this issue is related to a exit code 22 as well, it might be the same underlying issue: https://github.com/roopakv/orbs/issues/76

Reply:
Actually, there might be another issue. 
When running the current script in ssh mode, circleci api authentication fails, resulting in `job_status` and `job_number` variables having `""` value.
Which is equivalent to "the job we wait for doesn't exist", since we check `"$job_status" == ""`:
```
if [[ "$job_status" == "success" || "$job_status" == "failed" || "$job_status" == "canceled" || "$job_status" == "" ]]; then
    echo "Its finally my turn. exiting"
```
This is an issue that will be solved by proper authentication, but I cannot reproduce the code 22 error though.

Reply:
I think I found the reason for the code 22 exit status:
https://curl.se/mail/archive-2003-03/0079.html
```
--fail will return 22 for any HTTP error that is 400 or above
```

For some reason the `-s` option in `curl` command hid it, but it seems to be the culprit.
<img width="484" alt="Screenshot 2022-02-13 at 20 41 34" src="https://user-images.githubusercontent.com/1590371/153771862-34ca4cb3-44de-4b4d-b42b-2d7511f9923b.png">

Now, I've no idea:
- why it occurs on some configuration, but not on yours (could you check if authentication really works, or if the failing auth is just silently ignored, resulting in thinking the job we wait for doesn't exist?)
- why when running manually ssh command I don't get an exit code 22, but it appears only when running the regular CI workflows 🤷‍♂️

Reply:
so i double checked this. the PR in the current state will not work.

=> `Authorization` header does not work with circle token. 
=> to use the Circle token method you need to have header `Circle-Token`

i.e.

does not work
```
curl --location --request GET 'https://circleci.com/api/v2/pipeline?org-slug=gh/roopakv' \
--header 'Authorization: Basic TOKEN_HERE'
```

works
```
curl --location --request GET 'https://circleci.com/api/v2/pipeline?org-slug=gh/roopakv' \
--header 'Circle-Token: TOKEN_HERE'
```

Reply:
I don't have any preference so let's go with the `Circle-Token` header 👍 

It's strange though, for me it works in both cases 🤷‍♂️ 
<img width="1301" alt="Screenshot 2022-02-14 at 11 57 14" src="https://user-images.githubusercontent.com/1590371/153851341-d96a23bc-9a43-4c30-8cbe-8f89fe02f768.png">